src/Stratis.Bitcoin.Features.Consensus/Rules/ConsensusRule.cs

```diff
@@ -28,9 +27,14 @@ public virtual IEnumerable<Type> Dependencies()
         }
 
         /// <summary>
-        /// An indicator whether this rule is allowed to skip validation when the <see cref="BlockValidationContext.SkipValidation"/> is set to <c>true</c>.
+        /// Whether the rule will be considered a rule that only does validation and does not manipulate state in any way.
+        /// When <c>true</c> rule is allowed to skip validation when the <see cref="BlockValidationContext.SkipValidation"/> is set to <c>true</c>.
         /// </summary>
-        public virtual bool CanSkipValidation => false;
+        /// <remarks>
+        /// State in this context is the manipulation of information in the consensus data store based on actions specified <see cref="Block"/> and <see cref="Transaction"/>.
+        /// This will allow to ability to run validation checks on blocks (during mining for example) without change the underline store.
+        /// </remarks>
+        public virtual bool ValidationRule => true;
```

Comment:
Not very clear what this means.
Does it mean it's a rule that is subject to validation, or that it's a validating rule?
Maybe calling thia ValidationOnlyRule will make it clearer.

Reply:
Yeah I did actually called it that `ValidationOnlyRule` before and then was not happy with that name. 
This means the rules only does validation.
Hmm perhaps that name is good, any other suggestions?